### ISSUE TITLE
Incorrect histogram count and timerange shown in the histogram chart

### ISSUE THREAD

**Issue by ankitnayan** [SigNoz/signoz#5450]
The timestamps in the chart probably does not follow the aggregation interval chosen automatically by the time range

[Logs explorer](https://saasmonitor.us.signoz.cloud/logs/logs-explorer?options=%7B%22selectColumns%22%3A%5B%7B%22key%22%3A%22serviceName%22%2C%22dataType%22%3A%22string%22%2C%22type%22%3A%22tag%22%2C%22isColumn%22%3Atrue%2C%22isJSON%22%3Afalse%2C%22id%22%3A%22serviceName--string--tag--true%22%7D%2C%7B%22key%22%3A%22name%22%2C%22dataType%22%3A%22string%22%2C%22type%22%3A%22tag%22%2C%22isColumn%22%3Atrue%2C%22isJSON%22%3Afalse%2C%22id%22%3A%22name--string--tag--true%22%7D%2C%7B%22key%22%3A%22durationNano%22%2C%22dataType%22%3A%22float64%22%2C%22type%22%3A%22tag%22%2C%22isColumn%22%3Atrue%2C%22isJSON%22%3Afalse%2C%22id%22%3A%22durationNano--float64--tag--true%22%7D%2C%7B%22key%22%3A%22httpMethod%22%2C%22dataType%22%3A%22string%22%2C%22type%22%3A%22tag%22%2C%22isColumn%22%3Atrue%2C%22isJSON%22%3Afalse%2C%22id%22%3A%22httpMethod--string--tag--true%22%7D%2C%7B%22key%22%3A%22responseStatusCode%22%2C%22dataType%22%3A%22string%22%2C%22type%22%3A%22tag%22%2C%22isColumn%22%3Atrue%2C%22isJSON%22%3Afalse%2C%22id%22%3A%22responseStatusCode--string--tag--true%22%7D%5D%2C%22maxLines%22%3A2%2C%22format%22%3A%22list%22%7D&compositeQuery=%257B%2522queryType%2522%253A%2522builder%2522%252C%2522builder%2522%253A%257B%2522queryData%2522%253A%255B%257B%2522dataSource%2522%253A%2522logs%2522%252C%2522queryName%2522%253A%2522A%2522%252C%2522aggregateOperator%2522%253A%2522noop%2522%252C%2522aggregateAttribute%2522%253A%257B%2522id%2522%253A%2522------false%2522%252C%2522dataType%2522%253A%2522%2522%252C%2522key%2522%253A%2522%2522%252C%2522isColumn%2522%253Afalse%252C%2522type%2522%253A%2522%2522%252C%2522isJSON%2522%253Afalse%257D%252C%2522timeAggregation%2522%253A%2522rate%2522%252C%2522spaceAggregation%2522%253A%2522sum%2522%252C%2522functions%2522%253A%255B%255D%252C%2522filters%2522%253A%257B%2522items%2522%253A%255B%257B%2522id%2522%253A%25223ebf050b%2522%252C%2522key%2522%253A%257B%2522key%2522%253A%2522k8s.namespace.name%2522%252C%2522dataType%2522%253A%2522string%2522%252C%2522type%2522%253A%2522resource%2522%252C%2522isColumn%2522%253Atrue%252C%2522isJSON%2522%253Afalse%252C%2522id%2522%253A%2522k8s.namespace.name--string--resource--true%2522%257D%252C%2522op%2522%253A%2522%253D%2522%252C%2522value%2522%253A%2522primer%2522%257D%252C%257B%2522id%2522%253A%2522973d5c6e%2522%252C%2522key%2522%253A%257B%2522key%2522%253A%2522k8s.pod.name%2522%252C%2522dataType%2522%253A%2522string%2522%252C%2522type%2522%253A%2522resource%2522%252C%2522isColumn%2522%253Atrue%252C%2522isJSON%2522%253Afalse%252C%2522id%2522%253A%2522k8s.pod.name--string--resource--true%2522%257D%252C%2522op%2522%253A%2522in%2522%252C%2522value%2522%253A%255B%2522my-release-signoz-query-service-0%2522%255D%257D%252C%257B%2522id%2522%253A%2522f8448281%2522%252C%2522key%2522%253A%257B%2522key%2522%253A%2522severity_text%2522%252C%2522dataType%2522%253A%2522string%2522%252C%2522type%2522%253A%2522%2522%252C%2522isColumn%2522%253Atrue%252C%2522isJSON%2522%253Afalse%252C%2522id%2522%253A%2522severity_text--string----true%2522%257D%252C%2522op%2522%253A%2522in%2522%252C%2522value%2522%253A%255B%2522ERROR%2522%252C%2522FATAL%2522%255D%257D%255D%252C%2522op%2522%253A%2522AND%2522%257D%252C%2522expression%2522%253A%2522A%2522%252C%2522disabled%2522%253Afalse%252C%2522stepInterval%2522%253A60%252C%2522having%2522%253A%255B%255D%252C%2522limit%2522%253Anull%252C%2522orderBy%2522%253A%255B%257B%2522columnName%2522%253A%2522timestamp%2522%252C%2522order%2522%253A%2522desc%2522%257D%255D%252C%2522groupBy%2522%253A%255B%255D%252C%2522legend%2522%253A%2522%257B%257Bk8s.namespace.name%257D%257D%2522%252C%2522reduceTo%2522%253A%2522avg%2522%257D%255D%252C%2522queryFormulas%2522%253A%255B%255D%257D%252C%2522promql%2522%253A%255B%257B%2522name%2522%253A%2522A%2522%252C%2522query%2522%253A%2522%2522%252C%2522legend%2522%253A%2522%2522%252C%2522disabled%2522%253Afalse%257D%255D%252C%2522clickhouse_sql%2522%253A%255B%257B%2522name%2522%253A%2522A%2522%252C%2522legend%2522%253A%2522%2522%252C%2522disabled%2522%253Afalse%252C%2522query%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522da5c88e3-61e6-44c9-9585-574fddb6a7f0%2522%257D&panelTypes=%22list%22&viewName=%22%22&viewKey=%22%22&activeLogId=%222ixaXEQhacJ6gkCRKBqv57nDUOt%22&timeRange=%7B%22start%22%3A1720351986000%2C%22end%22%3A1720435147000%2C%22pageSize%22%3A11%7D&relativeTime=1d)
![Screenshot 2024-07-08 at 4 59 59 PM](https://github.com/SigNoz/signoz/assets/12460410/57a3513e-346a-4760-898a-2fa4397f0340)

**Comment by vikrantgupta25** [comment]
if we filter by last 30 minutes the list query has data but the graph query returns empty data.  CC :- @nityanandagohain 

and if we filter by the above said time range as well the graph api returns only data at 2 data points ( 10,1) which is shown above. 

`aggregation interval chosen automatically by the time range` :- this i need to check further if we rely on API response for the aggregations ? because what we pass in the payload might not be actually what has been used in the query. but will check and fill more details here

**Comment by nityanandagohain** [comment]
@vikrantgupta25 this has reveled a new bug which was hiding. If you open the network tab and see.

Two queries are fired, one for graph and one for list data.

The timerange passed for list API call sends a time range of 1 day when 30 mins timerange is selected.

**Comment by souravpurkait360** [comment]
`The timerange passed for list API call sends a time range of 1 day when 30 mins timerange is selected.`
I couldn't repro this one. Maybe this issue fixed already? Though the only issue I saw was that in logs explorer screen we are not retaining the start - end date time range. That also might be intentional because I saw the code changes by @vikrantgupta25 that we are particularly excluding the logs explorer for pushing the selected date time in the `urlQuery`. 

![Screenshot 2024-07-27 at 11 12 46 PM](https://github.com/user-attachments/assets/fb8effcb-6600-4218-9541-f8b02ee4c1c3)
